src/libcore/iter.rs

```diff
@@ -148,7 +148,7 @@ pub trait Iterator {
         last
     }
 
-    /// Loops through `n` iterations, returning the `n`th element of the
+    /// Loops through `n` iterations, returning the next (`n+1`th) element of the
```

Comment:
We shouldn't use a formulation with “n + 1 th”: the meaning of “nth” as used in this sentence is using a different convention than the `nth` in the method name. Best to word it differently.